specs/interop/predeploys.md

```diff
@@ -86,6 +90,142 @@ To ensure safety of the protocol, the [Message Invariants](./messaging.md#messag
 
 [`Identifier`]: ./messaging.md#message-identifier
 
+### Access-list
+
+Execution of messages is statically pre-declared in transactions,
+to ensure the cross-chain validity can be verified outside the single-chain EVM environment constraints.
+
+After pre-verification of the access-list, the `CrossL2Inbox` can allow messages
+to execute when there is a matching pre-verified access-list entry.
+
+Each executing message is declared with 3 typed access-list entries:
+- 1: Lookup identity
+- 2: ChainID extension
+- 3: Checksum
+
+The type of entry is encoded in the first byte.
+Type 0 is reserved, so valid access-list entries are always non-zero.
+
+Note that the access-list entries may be de-duplicated:
+the same message may be executed multiple times.
+
+The access-list content not always a multiple of 3.
+
+The access-list content is ordered:
+- after type 1, a type 2 or 3 entry is expected.
+- after type 2, a type 3 entry is expected.
+
+Note that type 1 and 2 are only enforced out-of-protocol:
+these provide a hint, for viable block-building,
+to lookup data to determine the validity of the checksum without prior transaction execution.
+
+Not every access-list entry may be executed:
+access-list content must not be used by applications to interpret results of transactions,
+the `ExecutingMessage` event describes in detail what is executed.
+
+To prevent cross-contamination of access-list contents,
+the checksum entry commits to the contents of the other entries.
+The checksum will be invalid if the wrong entries are interpreted with it.
+
+The `CrossL2Inbox` only checks the checksum is present in the access-list:
+the presence of other needed entries is enforced during pre-validation.
```

Comment:
This is really key, and keeps execution relatively cost efficient, since we only pay for 1 cold `SLOAD` per message.

Besides, block-builders that fill access-lists for themselves can optimize this way, and not include unnecessary access-list data. The checksum per message is enough in that case.